src/main/scala/diplomacy/Resources.scala

```diff
@@ -361,10 +371,34 @@ trait BindingScope
     ResourceMap(SortedMap("/" -> tree))
   }
 
+  /***
+    * Find the device's ResourceBindings in the ResourceBindingsMap
+    * @param dev
+    * @param rsb
+    */
+  def getResourceBindings(dev: Device, rsb: ResourceBindingsMap): Option[ResourceBindings] = {
+    if (rsb.map.contains(dev)) {
+      Some(rsb.map.lift(dev).getOrElse(ResourceBindings()))
+    }
+    else {
+      None
+    }
+  }
+
+  /** Generate the ResourceBindingsMap which stores each device's ResourceBindings
+    *
+    * @return
+    */
+  def getResourceBindingsMap: ResourceBindingsMap = {
+    eval
+    ResourceBindingsMap(map = resourceBindings.reverse.groupBy(_._1.owner).mapValues(seq => ResourceBindings(
+        seq.groupBy(_._1.key).mapValues(_.map(z => Binding(z._2, z._3)).distinct))))
```

Comment:
This is the same expression that is in the `bindingTree` method. Can you refactor it such that `bindingTree` calls this?

Reply:
removed